File: R/number_to_text.R
#inputnum <- readline("1-9999 number? ")
inputnum <- "222"

num2text <- function(x)
  {
    return(
      switch(
        x,
        "1" = "one",
        "2" = "two",
        "3" = "three",
        "4" = "four",
        "5" = "five",
        "6" = "six",
        "7" = "seven",
        "8" = "eight",
        "9" = "nine",
        "10" = "ten",
        "11" = "eleven",
        "12" = "twelve",
        "13" = "thirteen",
        "14" = "fourteen",
        "15" = "fifteen",
        "16" = "sixteen",
        "17" = "seventeen",
        "18" = "eighteen",
        "19" = "nineteen",
        "20" = "twenty",
        "30" = "thirty",
        "40" = "forty",
        "50" = "fifty",
        "60" = "sixty",
        "70" = "seventy",
        "80" = "eighty",
        "90" = "ninety"
      )
    )
  }

if (length(str(inputnum)) == 1) {inputnum = 000 & inputnum}
if (length(str(inputnum)) == 2) {inputnum = 00 & inputnum}
if (length(str(inputnum)) == 3) {inputnum = 0 & inputnum}

num1 <- as.integer(inputnum) %% 10
num2 <- (as.integer(inputnum) / 10) %% 10
num3 <- (as.integer(inputnum) / 100) %% 10
num4 <- (as.integer(inputnum) / 1000) %% 10

num1 <- as.integer(num1)
num2 <- as.integer(num2)
num3 <- as.integer(num3)
num4 <- as.integer(num4)

print(num1)
print(num2)
print(num3)
print(num4)

if (inputnum < 21) {print(num2text(inputnum))}
if (inputnum > 20 & num4 == 0 & num3 == 0 & num2 == 0 & num1 == 0) {print(num2text(inputnum))}
if (inputnum > 20 & num4 == 0 & num3 == 0 & num2 > 0 & num1 > 0) {paste(num2text(toString(num2 * 10)), num2text(toString(num1)))}
if (inputnum > 20 & num4 == 0 & num3 > 0 & num2 == 0 & num1 == 0) {paste(num2text(toString(num3)), "hundred")}
if (inputnum > 20 & num4 == 0 & num3 > 0 & num2 > 0 & num1 == 0) {paste(num2text(toString(num3)), "hundred and", num2text(toString(num2 * 10)))}
if (inputnum > 20 & num4 == 0 & num3 > 0 & num2 > 0 & num1 > 0) {paste(num2text(toString(num3)), "hundred and", num2text(toString(num2 * 10)), num2text(toString(num1)))}
# thousands code here...

File: Scala/number_to_text.scala
var num : String = "3891"

def num2text(num:Int):String=
	{
	num match
		{
		case 1 => "one"
        case 2 => "two"
        case 3 => "three"
        case 4 => "four"
        case 5 => "five"
        case 6 => "six"
        case 7 => "seven"
        case 8 => "eight"
        case 9 => "nine"
        case 10 => "ten"
        case 11 => "eleven"
        case 12 => "twelve"
        case 13 => "thirteen"
        case 14 => "fourteen"
        case 15 => "fifteen"
        case 16 => "sixteen"
        case 17 => "seventeen"
        case 18 => "eighteen"
        case 19 => "nineteen"
        case 20 => "twenty"
        case 30 => "thirty"
        case 40 => "forty"
        case 50 => "fifty"
        case 60 => "sixty"
        case 70 => "seventy"
        case 80 => "eighty"
        case 90 => "ninety"
		}
	}
	
if (num.length == 1) {num = "000" + num}
if (num.length == 2) {num = "00" + num}
if (num.length == 3) {num = "00" + num}

val num1 = (num.toInt % 10)
val num2 = (num.toInt / 10 % 10)
val num3 = (num.toInt / 100 % 10)
val num4 = (num.toInt / 1000 % 10)
val num2num1 = (num2.toString + num1.toString)

println("num1 - " + num1)
println("num2 - " + num2)
println("num3 - " + num3)
println("num4 - " + num4)
println("num2num1 - " + num2num1)

if (num.toInt < 21) {print(num2text(num2num1.toInt))}
if (num.toInt > 20 & num4 == 0 & num3 == 0 & num2 > 0 & num1 == 0) {print(num2text(num2 * 10))}
if (num.toInt > 20 & num4 == 0 & num3 == 0 & num2 > 0 & num1 > 0) {print(num2text(num2 * 10) + " " + num2text(num1))}
if (num.toInt > 20 & num4 == 0 & num3 > 0 & num2 == 0 & num1 == 0) {print(num2text(num3) + " hundred")}
if (num.toInt > 20 & num4 == 0 & num3 > 0 & num2 > 0 & num1 == 0) {print(num2text(num3) + " hundred and " + num2text(num2 * 10))}
if (num.toInt > 20 & num4 == 0 & num3 > 0 & num2 > 0 & num1 > 0 & num2num1.toInt < 21) {print(num2text(num3) + " hundred and " + num2text(num2num1.toInt))}
if (num.toInt > 20 & num4 == 0 & num3 > 0 & num2 > 0 & num1 > 0 & num2num1.toInt > 21) {print(num2text(num3) + " hundred and " + num2text(num2 * 10) + " " + num2text(num1))}
if (num.toInt > 20 & num4 > 0 & num3 == 0 & num2 == 0 & num1 == 0) {print(num2text(num4) + " thousand")}
if (num.toInt > 20 & num4 > 0 & num3 == 0 & num2 > 0 & num1 > 0 & num2num1.toInt < 21) {print(num2text(num4) + " thousand and " + num2text(num2num1.toInt))}
if (num.toInt > 20 & num4 > 0 & num3 == 0 & num2 == 0 & num1 > 0) {print(num2text(num4) + " thousand and " + num2text(num1))}
if (num.toInt > 20 & num4 > 0 & num3 > 0 & num2 == 0 & num1 == 0) {print(num2text(num4) + " thousand, " + num2text(num3) + " hundred")}
if (num.toInt > 20 & num4 > 0 & num3 > 0 & num2 > 0 & num1 == 0) {print(num2text(num4) + " thousand, " + num2text(num3) + " hundred and " + num2text(num2 * 10))}
if (num.toInt > 20 & num4 > 0 & num3 > 0 & num2 > 0 & num1 > 0) {print(num2text(num4) + " thousand, " + num2text(num3) + " hundred and " + num2text(num2 * 10) + " " + num2text(num1))}
